Library/Homebrew/rubocops/cask/extend/node.rbi

```diff
@@ -0,0 +1,21 @@
+# typed: strict
+
+class RuboCop::AST::Node
+  sig { returns(T::Boolean) }
+  def arch_variable?; end
+
+  sig { returns(T::Boolean) }
+  def begin_block?; end
+
+  sig { returns(T.nilable(RuboCop::AST::Node)) }
```

Comment:
```suggestion
  sig { returns(T.nilable(T.self_type)) }
```

Reply:
Tricky, but I believe this is correct as written. `T.self_type` returns the [receiver](https://sorbet.org/docs/self-type) type, not the declaring type. To validate this, I debugged the call `stanza_block`, which was `RuboCop::AST::BlockNode` for `block_node`, but `RuboCop::AST::Node` for `block_node.block_body`.

Reply:
Yes, makes sense.